pumps/mongo.go

```diff
@@ -258,16 +258,13 @@ func (m *MongoPump) connect() {
 		}).Panic("Mongo URL is invalid: ", err)
 	}
 
+	dialInfo.Timeout = time.Second * 5
 	m.dbSession, err = mgo.DialWithInfo(dialInfo)
 
-	// TODO - Should this not bail after a while?
-	for err != nil {
+	if err != nil {
 		log.WithFields(logrus.Fields{
 			"prefix": mongoPrefix,
-		}).Error("Mongo connection failed:", err)
-
-		time.Sleep(5 * time.Second)
```

Comment:
Previously it could auto-heal by itself since it was inside the loop, now it will fail on first run. Is that expected? Not sure we should change this behavior.

Reply:
well, I can change it to print error message and retry connection instead of failing?

Reply:
I fixed it!